ledger/src/blockstore_meta.rs

```diff
@@ -37,9 +46,8 @@ pub struct SlotMeta {
     // The list of slots, each of which contains a block that derives
     // from this one.
     pub next_slots: Vec<Slot>,
-    // True if this slot is full (consumed == last_index + 1) and if every
-    // slot that is a parent of this slot is also connected.
-    pub is_connected: bool,
+    // True if this slot's parent is connected
+    pub connected_flags: ConnectedFlags,
```

Comment:
@CriesofCarrots - We've talked about this / I know you've dealt with this kind of issue in the past, so was hoping I could get a sanity check on this. If you want more context as to why I'm messing with this at all, check out https://github.com/solana-labs/solana/issues/27820; otherwise, skip ahead to following paragraph.

I'm trying to add some extra info to the `SlotMeta` struct. So, I'm using bitflags to write a u8 in-place of the bool that was previously there that was also occupying a full byte.
- The tracking `is_connected` provides is currently broken (this bit is always false), but even if it weren't, the case where someone upgrades their client without wiping their ledger would work
- However, the downgrade case is a little more problematic. If a user runs with a new client (that sets multiple bits in this byte) and then reverts to a client that is setting a bool, deserialize will fail, complaining about an illegal value for the bool.

So, my thinking is that I could roll out the following commit in this order:
- Change master to use bitflag in place of bool
   - This change would only be for the datatype (ie no is_parent option YET)
- Backport this to our relevant release branches (I guess v1.14 / v1.13 / v1.10)
- Once all clusters are safely using versions that have the bitflags, introduce the functionality that comes with `is_parent` option
   - This is the point at which being able to deserialize into a bool (ie largescale rollback) is no longer works
   
Just want to make sure this seems like a sane plan - if so, I'll then bring this up in #releng channel to see if I can get everyone on board with that. I know we're in a strange spot with releases right now

Reply:
> Dropping this in place of the existing bool is backwards compatible with the serialized SlotMeta so no worries about compatibility breaks.

this isn't _quite_ true.  rust strictly defines `bool::false` to be `0` and `bool::true` to be `1`. so it'll fit in the same space, but bool deserialize will blow up if extra bits are set

Reply:
Agh, I didn't word that very well. What I meant to indicate is that this code will run with a ledger that was populated setting only the one bit as a bool. Ie, if this land in v1.15, it will work with a ledger that was filled using v1.14. 

Using the same assumption as above that this lands in v1.15, if someone has a ledger running with v1.15 and then tries to downgrade their client to v1.14, deserialize will blow up.

The "stub" solution I proposed here is meant to provide a way to run v1.14 on a ledger that was filled by v1.15

Reply:
Or rather, I made statement in the PR description before I experimentally learned that deserialize will complain about extra bits being set 😄

Reply:
an alternative to changing the 1.14 layout could be to instead write a custom deserializer for that field that accepts the new format, only if valid, and discards any new format bits before emitting a `bool`